### ISSUE TITLE
Checksum mismatch error not helpful

### ISSUE THREAD

**Issue by o2dac** [flyway/flyway#1813]
###### What version of Flyway are you using?
3.2.1

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin, SBT plugin, ANT tasks)
Spring boot, Maven

###### What database are you using (type & version)?
Postgres 9.3

###### What operating system are you using?
Ubuntu 16.04

###### What did you do?
Developer (e.g. accidentally) changed one of the migration scripts file.

###### What did you expect to see?
Flyway should log the error along with file name, so we can quickly identify what file was changed and check the history of the file.

###### What did you see instead?
It logged only the checksums. So if we want to troubleshoot the migration script we need to go to the `schema_version` table and looking for the checksums.
Below is an excerpt of what was printed on the stack trace
```
org.flywaydb.core.api.FlywayException: Validate failed. Migration Checksum mismatch for migration 2
-> Applied to database : 168520745
-> Resolved locally    : -1886360926
```

**Comment by axelfontaine** [comment]
Well isn't it trivial to find this out based on the version number provided by the error message?

**Comment by o2dac** [comment]
Unfortunately, I didn’t see the version mentioned anywhere in the error log. Does “migration 2” refer to “version” that you’re talking about? If true, that’s a bit clearer, but it means something else went wrong, because the version that actually causes the checksum mismatch error is version 19 in my configuration. So yes, a vesion info is good enough, as long as it is actually printed out. E.g. if my file is named V1.8__some_change.sql, then the error log should at least say “V1.8”. Currently it doesn’t.

**Comment by axelfontaine** [comment]
Even though the message already contained the version number for versioned migrations, we decided the name making in even more explicit by also including the string `version`. The message you got is now printed as

```
org.flywaydb.core.api.FlywayException: Validate failed. Migration Checksum mismatch for migration version 2
-> Applied to database : 168520745
-> Resolved locally    : -1886360926
```
